src/env/env.rs

```diff
@@ -188,8 +188,19 @@ pub fn uumain(args: Vec<String>) -> i32 {
     }
 
     if !opts.program.is_empty() {
-        let prog = opts.program[0].clone();
-        let args = &opts.program[1..];
+        let prog;
+        let args;
+        { #![cfg(not(windows))]
```

Comment:
I think it would be better to make these two platform-dependent functions that return `prog` and `args` in a tuple and take `opts` as an argument.  Then you can just do `let (prog, args) = func(&mut opts);` or something like that.  By the way, you might want to look into make `prog` use `Cow` to avoid allocating when not needed (_e.g._ when `env::var("ComSpec")` fails to give a value).

Reply:
I'll look into it.

Reply:
I finally had some time to refactor this for you (see <https://github.com/uutils/coreutils/pull/1316/commits/75a163d9ba6e45bc7dbbd36c6363bff5b0abba1a>).

While I was modifying the code, I took the time to add a new "--file" feature which can read additional variable settings from text files (see <https://github.com/uutils/coreutils/pull/1316/commits/0807f60a8a5cc3120877e56009c48fedd3602af5>).

If you're happy with the final product, I'll squash and rebase it onto your master (I see there was some code change to `env` in recent commits).